packages/framework/src/Console/Commands/MakePublicationTypeCommand.php

```diff
@@ -109,41 +112,69 @@ protected function captureFieldsDefinitions(): Collection
             $this->output->writeln("<bg=cyan;fg=white>Field #$count:</>");
 
             $field = Collection::create();
-            $field->name = $this->askWithValidation('name', 'Field name', ['required']);
+            do {
+                $field->name = trim($this->askWithValidation('name', 'Field name', ['required']));
+                $duplicate   = $fields->where('name', $field->name)->count();
+                if ($duplicate) {
+                    $this->error("Field name [$field->name] already exists!");
+                }
+            } while ($duplicate);
+
             $this->line('Field type:');
             $this->line('  1 - String');
             $this->line('  2 - Boolean ');
             $this->line('  3 - Integer');
             $this->line('  4 - Float');
-            $this->line('  5 - Datetime');
+            $this->line('  5 - Datetime (YYYY-MM-DD (HH:MM:SS))');
             $this->line('  6 - URL');
             $this->line('  7 - Array');
             $this->line('  8 - Text');
             $this->line('  9 - Local Image');
-            $type = (int) $this->askWithValidation('type', 'Field type (1-9)', ['required', 'integer', 'between:1,9'], 1);
-            do {
-                // TODO This should only be done for types that can have length restrictions right?
-                $field->min = $this->askWithValidation('min', 'Min value (for strings, this refers to string length)', ['required', 'string'], 0);
-                $field->max = $this->askWithValidation('max', 'Max value (for strings, this refers to string length)', ['required', 'string'], 0);
-                $lengthsValid = true;
-                if ($field->max < $field->min) {
-                    $lengthsValid = false;
-                    $this->output->warning('Field length [max] must be [>=] than [min]');
+            $this->line('  10 - Tag (select value from list)');
+            $type = (int) $this->askWithValidation('type', 'Field type (1-10)', ['required', 'integer', 'between:1,10'], 1);
+
+            if ($type < 10) {
+                do {
+                    // TODO This should only be done for types that can have length restrictions right?
+                    // ANSWER: No, it can also be a value restriction
+                    // - (int: 0 - 2022)
+                    // - (float: 0 - 360)
+                    // - (datetime: 2022-01-01 - 2022-12-31)
```

Comment:
Oh yeah right, thanks! Question: are there any types that can't have length restrictions? How about images?

Reply:
Currently, every field except tags asks for a length ... this should/could be refined (I'll take care of it). 

The only problem is that it's context-dependent. For strings, it's the string length, for numbers it's the numeric value, for dates it's the date range (which isn't validated properly yet), so let me take a look and see how to best do this.

Reply:
Okay, sounds great. Thanks!